d3/d3.d.ts

```diff
@@ -119,7 +119,7 @@ declare module D3 {
         * @param arr Array to search
         * @param map Accsessor function
         */
-        extent<T, U>(arr: T[], map: (v: T) => U): U[];
+        extent<T, U>(arr: T[], map: (v: T, i?: number) => U): U[];
```

Comment:
This is probably true for the other map functions below, but the map function here also has a third parameter which contains the entire array, I believe.  Also, this map function technically has all 3 parameters be optional.